warehouse/manage/forms.py

```diff
@@ -61,10 +61,20 @@ class AddEmailForm(NewEmailMixin, forms.Form):
 
     __params__ = ["email"]
 
-    def __init__(self, *args, user_service, **kwargs):
+    def __init__(self, *args, request, user_service, **kwargs):
         super().__init__(*args, **kwargs)
+        self.request = request
         self.user_service = user_service
 
+    def validate_email(self, field):
+        userid = self.user_service.find_userid_by_email(field.data)
+        if userid is not None and userid == self.request.user.id:
+            raise wtforms.validators.ValidationError(
+                "This email address is already being used by this account. "
+                "Use a different email."
+            )
+        return super().validate_email(field)
```

Comment:
I think this check should be part of the `validate_email` function of the `NewEmailMixin`, rather than being overridden here.